### PR TITLE
create form for adding project helm chart repo

### PR DESCRIPTION
**Pull request by debsmita1**
**Jira Story:**
https://issues.redhat.com/browse/ODC-6517

**Root analysis:**
As a user, I want to use a form to add helm chart repository in my namespace

**Solution description:**
- Added name, repo URL , description fields
- Added checkbox to allow users to enable/disable the usage of the chart repository
- Added Secrets and ConfigMaps dropdowns in an Expand/Collapsible section for the user to add private helm chart repositories, the values in these dropdowns are fetched from the `openshift-config` namespace
- Submitting the form creates ProjectHelmChartRepository CR and redirects the user to the helm catalog page

**GIF:**
create-phcr

**Test setup:**
Goto route `/ns/<namespace>/project-helm-chart-repository/~new`

### CODE REVIEW

File: frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryForm.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+}) => {
+  const { t } = useTranslation();
+  const autocompleteFilter = (strText, item): boolean => fuzzy(strText, item?.props?.name);
+
+  return (
+    <FlexForm onSubmit={handleSubmit}>
+      <FormBody>
+        <FormHeader
+          title="Create ProjectHelmChartRepository"
+          helpText="Add helm chart repository in the namespace"
```
Comment (rottencandy) [inline frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryForm.tsx:35]: Needs i18n.

File: frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/projecthelmchartrepository-validation-utils.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { TFunction } from 'i18next';
+import * as yup from 'yup';
+import { nameRegex } from '@console/shared/src';
+
+const urlRegex = /^((https?:?):\/\/:?)/;
```
Comment (rottencandy) [inline frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/projecthelmchartrepository-validation-utils.ts:5]: Don't think we need the capture groups if we aren't using them.
And can also check for text after https
https://regex101.com/r/g2k6tk/1
```suggestion
const urlRegex = /^https?:?:\/\/:?.+/;
```

**Review by rottencandy — APPROVED** [review]
/lgtm

File: frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryForm.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+              />
+            </FormSection>
+          </ExpandCollapse>
+        </FormSection>
+      </FormBody>
+      <FormFooter
+        handleReset={handleReset}
+        errorMessage={status?.submitError}
+        isSubmitting={isSubmitting}
+        submitLabel="Create"
```
Comment (karthikjeeyar) [inline frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryForm.tsx:118]: Translation missing
```suggestion
        submitLabel=t('helm-plugin~Create')
```

File: frontend/packages/helm-plugin/console-extensions.json [hunk 4]
```diff
@@ -97,6 +97,14 @@
       "required": ["OPENSHIFT_HELM"]
     }
   },
+  {
+    "type": "console.page/route",
+    "properties": {
+      "exact": true,
+      "path": ["/ns/:ns/project-helm-chart-repository"],
```
Comment (karthikjeeyar) [inline frontend/packages/helm-plugin/console-extensions.json:104]: I think we should stick to same route convention that we have for other resources. WDYT?

```suggestion
      "path": ["/ns/:ns/projecthelmchartrepositories/~new"],
```
cc: @invincibleJai

File: frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryPage.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { Formik } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { match as RMatch } from 'react-router-dom';
+import { k8sCreateResource } from '@console/dynamic-plugin-sdk/src/utils/k8s';
+import { ProjectHelmChartRepositoryModel } from '@console/helm-plugin/src/models';
```
Comment (invincibleJai) [inline frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryPage.tsx:6]: nit: as `ProjectHelmChartRepositoryModel` is in same package so can do relative import

File: frontend/packages/helm-plugin/console-extensions.json [hunk 6]
```diff
@@ -97,6 +97,14 @@
       "required": ["OPENSHIFT_HELM"]
     }
   },
+  {
+    "type": "console.page/route",
+    "properties": {
+      "exact": true,
+      "path": ["/ns/:ns/project-helm-chart-repository/~new"],
```
Comment (karthikjeeyar) [inline frontend/packages/helm-plugin/console-extensions.json:104]: should this be plural?
```suggestion
      "path": ["/ns/:ns/projecthelmchartrepositories/~new"],
```

**Comment by karthikjeeyar** [comment]
For console developer users (with limited permissions on  other namespaces) will not able to pull the secrets from `openshift-config` namespace, for them the secrets dropdown will be empty always. Is it possible to have secret in your current namespace also? and list it in this dropdown so you can access and reference the secrets that you own.


image

**Comment by karthikjeeyar** [comment]
This is a Project helm chart repo creation page, so should we include the project selector on the top of the page (namespaced page)?

File: frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryForm.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+          </ExpandCollapse>
+        </FormSection>
+      </FormBody>
+      <FormFooter
+        handleReset={handleReset}
+        errorMessage={status?.submitError}
+        isSubmitting={isSubmitting}
+        submitLabel={t('helm-plugin~Create')}
+        disableSubmit={isSubmitting || !_.isEmpty(errors)}
+        resetLabel={t('helm-plugin~Cancel')}
```
Comment (karthikjeeyar) [inline frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryForm.tsx:120]: Clicking on Cancel button, enables the `Submit` button even though the required fields are not filled.

**Comment by debsmita1** [comment]
> This is a Project helm chart repo creation page, so should we include the project selector on the top of the page (namespaced page)?

@karthikjeeyar This is just the form component, I will create a follow-up story when the UX is finalized

File: frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryForm.tsx (hunk 7 above)
Comment (debsmita1) [reply frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryForm.tsx:120]: Fixed it

**Comment by debsmita1** [comment]
> For console developer users (with limited permissions on other namespaces) will not able to pull the secrets from `openshift-config` namespace, for them the secrets dropdown will be empty always. Is it possible to have secret in your current namespace also? and list it in this dropdown so you can access and reference the secrets that you own.
> 
> image

@karthikjeeyar the namespace name is hardcoded here https://github.com/openshift/console/blob/master/pkg/helm/chartproxy/repos.go#L41 . Spoke to David and raised a bugzilla ticket https://bugzilla.redhat.com/show_bug.cgi?id=2071792

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Review by rottencandy — APPROVED** [review]
Verified on cluster.
/label qe-approved

**Comment by debsmita1** [comment]
Propagating docs and px acks from the Epic
/label docs-approved
/label px-approved

**Pull request merged**